rust-sdk/whirlpool/src/token.rs

```diff
@@ -333,4 +342,233 @@ mod tests {
         assert_eq!(mint_c, mint1);
         assert_eq!(mint_d, mint2);
     }
+
+    #[tokio::test]
+    #[serial]
+    async fn test_token_2022_extensions() {
+        let ctx = RpcContext::new().await;
+
+        // Create Token-2022 mint with transfer fee
+        let mint_te = setup_mint_te_fee(&ctx).await.unwrap();
+        let mint_account = ctx.rpc.get_account(&mint_te).await.unwrap();
+
+        // Test transfer fee at epoch 0
+        let older = get_current_transfer_fee(Some(&mint_account), 0).unwrap();
+        assert_eq!(older.fee_bps, 100); // 1%
+        assert_eq!(older.max_fee, 1_000_000_000); // 1 token
+
+        // Test transfer fee at epoch 2
+        let newer = get_current_transfer_fee(Some(&mint_account), 2).unwrap();
+        assert_eq!(newer.fee_bps, 150); // 1.5%
+        assert_eq!(newer.max_fee, 1_000_000_000); // 1 token
+
+        // Test with no fee
+        let no_fee_result = get_current_transfer_fee(None, 0);
+        assert!(no_fee_result.is_none());
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn test_token_2022_account() {
+        let ctx = RpcContext::new().await;
+
+        // Create basic Token-2022 mint (without transfer fee)
+        let mint = setup_mint_te(&ctx, &[]).await.unwrap();
+        let mint_account = ctx.rpc.get_account(&mint).await.unwrap();
+
+        // Verify account data
```

Comment:
Does this test any of the sdk functions or only the test_utils function? If the second don't think we actually need this test since BanksClient will just fail if the instructions in `setup_mint_te` are bad

Reply:
Great point, it is testing only utils function indeed. Updated to make work for actual functionality.